github-actions/deploy-previews/pack-and-upload-artifact/inject-artifact-metadata.js

```diff
@@ -0,0 +1,34 @@
+
+import {createRequire as __cjsCompatRequire} from 'module';
+const require = __cjsCompatRequire(import.meta.url);
+
+
+// 
+import path from "path";
+import fs from "fs";
+
+// 
+var artifactMetadata = {
+  "pull-number": "./__metadata__pull_number.txt",
+  "build-revision": "./__metadata__build_revision.txt"
+};
+
+// 
+async function main() {
+  const [deployDirPath, prNumber, buildRevision] = process.argv.slice(2);
+  await fs.promises.writeFile(path.join(deployDirPath, artifactMetadata["pull-number"]), prNumber);
+  await fs.promises.writeFile(path.join(deployDirPath, artifactMetadata["build-revision"]), buildRevision);
+}
+try {
+  await main();
+} catch (e) {
+  console.error(e);
+  process.exit(1);
+}
+/**
```

Comment:
Can this be at the top of the file?

Reply:
Same comment as above. This is the generated file